### PR TITLE
Add NSP Labels Handling to DataCollatorForWholeWordMask for Simultaneous WWM and NSP Pre-training

### PR DESCRIPTION
**Pull request by callanwu**
# What does this PR do?

Integrating NSP with WWM allows for a more comprehensive pre-training of Encoder-only models, leveraging the benefits of both tasks to improve model performance and robustness.

- Modified `torch_call`, `tf_call`, and `numpy_call` methods in `DataCollatorForWholeWordMask` to process NSP labels.
- Updated input handling to include `next_sentence_label` and `token_type_ids` if present in the input examples.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Hey! Would you mind adding a little bit of documentation about why people should use this, how to use it and a small test? 🤗
